src/pages/api/public/traces.ts

```diff
@@ -88,11 +90,25 @@ export default async function handler(
         });
         res.status(200).json(newTrace);
       } else {
-        const newTrace = await prisma.trace.create({
-          data: {
-            id: id ?? undefined,
-            timestamp: new Date(),
+        const internalId = id ?? uuidv4();
+
+        const newTrace = await prisma.trace.upsert({
+          where: {
+            id: internalId,
+            projectId: authCheck.scope.projectId,
+          },
+          create: {
+            id: internalId,
             projectId: authCheck.scope.projectId,
+            timestamp: new Date(),
+            name: name ?? undefined,
+            userId: userId ?? undefined,
+            metadata: metadata ?? undefined,
+            release: release ?? undefined,
+            version: version ?? undefined,
+          },
+          update: {
+            timestamp: new Date(),
```

Comment:
Suggestion: remove timestamp from create and update; set by database on create and should not be updated when upserting the trace (unexpected behavior)